softlayer/resource_softlayer_dns_domain.go

```diff
@@ -197,45 +93,77 @@ func resourceSoftLayerDnsDomainRead(d *schema.ResourceData, meta interface{}) er
 
 	// retrieve remote object state
 	dns_domain, err := service.Id(dnsId).Mask(
-		"id,name,serial,updateDate," +
-			"resourceRecords[data,domainId,expire,host,minimum,mxPriority,refresh,retry,ttl,type]",
+		"id,name,updateDate,resourceRecords",
 	).GetObject()
 	if err != nil {
 		return fmt.Errorf("Error retrieving Dns Domain %d: %s", dnsId, err)
 	}
 
 	// populate fields
-	d.Set("id", *dns_domain.Id)
 	d.Set("name", *dns_domain.Name)
-	d.Set("serial", *dns_domain.Serial)
-	d.Set("update_date", *dns_domain.UpdateDate)
-	d.Set("records", read_resource_records(dns_domain.ResourceRecords))
+	if dns_domain.Serial != nil {
+		d.Set("serial", *dns_domain.Serial)
+	}
 
-	return nil
-}
+	if dns_domain.UpdateDate != nil {
+		d.Set("update_date", *dns_domain.UpdateDate)
+	}
 
-func read_resource_records(list []datatypes.Dns_Domain_ResourceRecord) []map[string]interface{} {
-	records := make([]map[string]interface{}, 0, len(list))
-	for _, record := range list {
-		r := make(map[string]interface{})
-		r["record_data"] = *record.Data
-		r["domain_id"] = *record.DomainId
-		r["expire"] = *record.Expire
-		r["host"] = *record.Host
-		r["minimum_ttl"] = *record.Minimum
-		r["mx_priority"] = *record.MxPriority
-		r["refresh"] = *record.Refresh
-		r["retry"] = *record.Retry
-		r["ttl"] = *record.Ttl
-		r["record_type"] = *record.Type
-		records = append(records, r)
+	// find a record with host @; that will have the current target.
+	for _, record := range dns_domain.ResourceRecords {
+		if *record.Type == "a" && *record.Host == "@" {
+			d.Set("target", *record.Data)
+			break
+		}
 	}
-	return records
+
+	return nil
 }
 
 func resourceSoftLayerDnsDomainUpdate(d *schema.ResourceData, meta interface{}) error {
-	// TODO - update is not supported - implement delete-create?
-	return fmt.Errorf("Not implemented. Update Dns Domain is currently unsupported")
+	// If the target has been updated, find the corresponding dns record and update its data.
+	sess := meta.(*session.Session)
+	domainId, _ := strconv.Atoi(d.Id())
+
+	if !d.HasChange("target") {
+		return nil
+	}
```

Comment:
This check probably isn't necessary in the first place, since `target` is the only field for which the schema allows an update (all other fields are either computed or `ForceNew: true`).  

But if this is to guard against some unforeseen logic, then it may be appropriate to at least return a non-nil `error`.  Otherwise, this is essentially a silent fail.